### ISSUE TITLE
Multiline promtps override previous line when exiting Neovim

### ISSUE THREAD

**Issue by zDonik1** [nushell/nushell#12104]
### Describe the bug

When extitng Neovim, if the custom prompt returned is more than 1 line, the last line of the new prompt is overriden, then recreating the prompt.

### How to reproduce

1. Set the prompt function to return a multi-line prompt: `$env.PROMPT_COMMAND = {|| "nushell\n--" }`
2. Restart nushell
3. Open neovim and exit
4. Start typing a new command (eg. nvim)

### Expected behavior

A prompt is not skipped, and the last line is not deleted/overridden.

### Screenshots

![image](https://github.com/nushell/nushell/assets/25223592/56ff3558-36c6-4521-9692-4aaee34f8d66)

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.90.1                                   |
| branch             |                                          |
| commit_hash        | c2992d5d8b8541b64e9fc1440ed2917fc9b757f7 |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.73.0 (cc66ad468 2023-10-03)      |
| rust_channel       | 1.73.0-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.73.0 (9c4383fb5 2023-08-26)      |
| build_time         | 2024-02-07 02:25:09 +00:00               |
| build_rust_channel | release                                  |
| allocator          | mimalloc                                 |
| features           | default, sqlite, trash, which, zip       |
| installed_plugins  |                                          |

**Comment by fdncred** [comment]
Does the same thing happen on 0.91.0?

**Comment by zDonik1** [comment]
Didn't notice an update came out, I installed nu a few days ago.
Anyhow, the problem still persists:

![image](https://github.com/nushell/nushell/assets/25223592/438356d7-a940-47f2-93b3-23d5133d533c)

the entire config.nu is just `$env.PROMPT_COMMAND = {|| "nushell\n--" }`.

And here is the table again:

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.91.0                                   |
| branch             |                                          |
| commit_hash        | 3016d7a64ccb2c2eac9f735f6144fc896ea724a5 |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.74.1 (a28077b28 2023-12-04)      |
| rust_channel       | 1.74.1-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.74.1 (ecb9851af 2023-10-18)      |
| build_time         | 2024-03-05 20:48:22 +00:00               |
| build_rust_channel | release                                  |
| allocator          | mimalloc                                 |
| features           | default, sqlite, trash, which, zip       |
| installed_plugins  |                                          |

**Comment by fdncred** [comment]
I don't have this problem, but my prompt is different. I set my newline in the prompt indicator instead of the prompt.
```nushell
# prompt
use modules\prompt\oh-my.nu git_prompt
$env.PROMPT_COMMAND = {|| (git_prompt).left_prompt }
$env.PROMPT_COMMAND_RIGHT = {|| (git_prompt).right_prompt }
$env.PROMPT_INDICATOR = {|| if ($env.LAST_EXIT_CODE == 0) {$"(ansi green_bold)\n❯ (ansi reset)"} else {$"(ansi red_bold)\n❯ (ansi reset)"}}
$env.TRANSIENT_PROMPT_COMMAND = ""
$env.TRANSIENT_PROMPT_COMMAND_RIGHT = {|| (git_prompt).right_prompt }
$env.TRANSIENT_PROMPT_INDICATOR = {|| if ($env.LAST_EXIT_CODE == 0) {$"(ansi light_yellow_bold)❯ (ansi reset)"} else {$"(ansi light_red_bold)❯ (ansi reset)"}}
```

However, I can reproduce your experience with `nu -n` and then `$env.PROMPT_COMMAND = {|| "nushell\n--" }`. I'm not sure why it's redrawing the prompt. I'm guessing this is a reeline problem.

**Comment by zDonik1** [comment]
I personally use Starship's default configuration for prompts, and their prompts start with a new line. Thanks for checking the issue out.

**Comment by Mrhyuan** [comment]
Yes same here. I can reproduce it as well. Im on Windows 11 using alacritty 0.91.0 + nushell + starship + neovim.

**Comment by sibouras** [comment]
this also happens when the terminal is resized, where's a workaround:`$env.PROMPT_COMMAND = {|| print -n 'nushell--' }`
